### PR TITLE
[Ninja] Update to 1.10

### PR DESCRIPTION
**Pull request by MVoz**

### CODE REVIEW

File: scripts/vcpkgTools.xml [hunk 1]
```diff
@@ -87,25 +87,25 @@
...
+        <version>1.10.0</version>
         <exeRelativePath>ninja</exeRelativePath>
-        <url>https://github.com/ninja-build/ninja/releases/download/v1.8.2/ninja-mac.zip</url>
-        <sha512>acadfb286eb7d93676629701917fa0c3c39f36daa068c169e4a098c29f97380d1ea95abfd42b04798ff118fd9dc93fdb250fcda36086bac20bc5506354214fc3</sha512>
-        <archiveName>ninja-mac-1.8.2.zip</archiveName>
+        <url>https://github.com/ninja-build/ninja/releases/download/v1.10.0/ninja-mac.zip</url>
+        <sha512>619a1924067a0b30fc5f8887f868d3ee5481838d2f0f158d031f7614a2a10b95a73d4a56b658d5d560283ebf809e2e536b968c6c01ff0108075c3f393f5780ba</sha512>
+        <archiveName>ninja-mac-1.10.0.zip</archiveName>
     </tool>
     <tool name="ninja" os="freebsd">
```
Comment (NancyLi1013) [inline scripts/vcpkgTools.xml:110]: Why are the versions and `sha512` on `freebsd` not updated?

**Review by NancyLi1013 — APPROVED** [review]

**Comment by NancyLi1013** [comment]
Related issue #10369

File: scripts/vcpkgTools.xml (hunk 1 above)
Comment (Voskrese) [reply scripts/vcpkgTools.xml:110]: > 
> 
> Why are the versions and `sha512` on `freebsd` not updated?

no https://github.com/ivysnow/ninja/releases latest 1.82

Comment (NancyLi1013) [reply scripts/vcpkgTools.xml:110]: Thanks for your update.

**Comment by NancyLi1013** [comment]
Hi @Voskrese 
Thanks for your PR.
LGTM now. 
Need to rerun this PR once the regressions on Windows are solved.

**Comment by ras0219-msft** [comment]
Thanks for the PR!

**Review by SNiTEBoBy — COMMENTED** [review]
Merge pull requested

**Pull request merged**